### PR TITLE
Add custom protocol 2

### PR DESCRIPTION
**Pull request by zvin**

### CODE REVIEW

**Comment by thundron** [comment]
@balena-ci retest

**Comment by zvin** [comment]
@balena-ci retest

File: lib/gui/etcher.ts [hunk 1]
```diff
@@ -54,6 +57,46 @@ async function checkForUpdates(interval: number) {
 	}
 }
 
+function getCommandLineURL(argv: string[]): string | undefined {
+	if (platform() === 'darwin') {
+		// macOS uses open-url events instead
+		return;
```
Comment (thundron) [inline lib/gui/etcher.ts:63]: I looked a bit into it and I was partially wrong so I apologise, but the `open-url` is the event used when opening an URL. When doing something like `./balenaEtcher\ 1.5.97/balenaEtcher.app/Contents/MacOS/balenaEtcher miao.img` (so calling the executable directly) it actually uses argv

Comment (thundron) [reply lib/gui/etcher.ts:63]: Another strange thing is that the error about the `-psn_...` parameter didn't pop-up with a package I built locally to test the above (removing the check)

**Pull request merged**